Fixes 38343: review Contribute page

d1nuc0m opened the pull request:
Contribute:

* Switch to one-sentence-per-line style
* Review "Preconfigured virtual machine" and "Setting up from git manually" subsections to up-to-date software

Notes:

- Vagrant section hasn't been tested but updated from docs
- Plugins and smart proxy sections haven't been reviewed

d1nuc0m committed:
Fixes 38343: review Contribute page

* Review "Preconfigured virtual machine" and "Setting up from git
  manually" subsections to up-to-date software

ekohl commented:
> Limit to 80 columns

Please do any reformatting in a separate commit so it's easier to see actual content changes.

We've adopted the one sentence per line in foreman-documentation and I prefer that over 80 columns for text. https://jeffkreeftmeijer.com/one-sentence-per-line/ explains some benefits.

d1nuc0m committed:
style: "Contribute" page 'one sentence per line'

d1nuc0m commented:
> > Limit to 80 columns
> 
> Please do any reformatting in a separate commit so it's easier to see actual content changes.

Done now @ekohl , sorry for the lots of force push

> We've adopted the one sentence per line in foreman-documentation and I prefer that over 80 columns for text. https://jeffkreeftmeijer.com/one-sentence-per-line/ explains some benefits.

As long as there is a standard it's fine for me :D

By the way, the link http://ci.theforeman.org/view/Foreman%20pipeline/job/test_develop in the docs now returns a 404, where could I find the supported Ruby versions on Jenkins now?

ekohl commented:
> By the way, the link http://ci.theforeman.org/view/Foreman%20pipeline/job/test_develop in the docs now returns a 404, where could I find the supported Ruby versions on Jenkins now?

https://github.com/theforeman/foreman/blob/develop/.github/matrix.json would be the authoritative source now.

d1nuc0m committed:
docs: sw versions source in Contribute

Replace older authoritative source for software versions in "Contribute"
page, align suggested modules to those in matrix.json

d1nuc0m commented:
> https://github.com/theforeman/foreman/blob/develop/.github/matrix.json would be the authoritative source now.

Thank you, done as an additional commit to also align sw versions in the page to those in matrix.json

d1nuc0m commented:
Updated modules after checking instructions again

ekohl commented on contribute.md:
```diff
@@ -97,48 +97,52 @@ In general, any Ruby version supported by Rails version required by Foreman will
+# (OPTIONAL) Change Ruby module version before installing
+dnf module enable ruby:3.3
+# (OPTIONAL) Switch Ruby module version if packages are already installed
+dnf module switch-to ruby:3.3
```
At this moment Foreman itself doesn't support Ruby 3.3. We currently use the base version in EL9 so you can leave out the DNF modularity

d1nuc0m replied:
Added a note for "future proofing" so that when it will be needed there already is a reminder that Ruby is shipped as a module

d1nuc0m replied:
Highlighted it is possible but unsupported

ekohl commented on contribute.md:
```diff
@@ -97,48 +97,52 @@ In general, any Ruby version supported by Rails version required by Foreman will
 In general, the latest stable Node.js should work. However if you are just starting out, you might want to develop against one of the versions that we test against to reduce the chances of you hitting an unexpected issue, although this is not required.
 
-You can view our [Github Actions configuration](https://github.com/theforeman/foreman/blob/develop/.github/workflows/js_tests.yml) to see supported NodeJS versions. Any version from that list will do. For populating the node_modules folder, npm version 4 or later is required. Alternatively, [yarn](https://yarnpkg.com) can also be used.
+You can view our [package.json configuration](https://github.com/theforeman/foreman/blob/develop/package.json) to see supported NodeJS versions. Any version from that list will do. For populating the node_modules folder, npm version 4 or later is required. Alternatively, [yarn](https://yarnpkg.com) can also be used.
```
I think we now only support NPM 6+ and don't think we really support yarn

d1nuc0m replied:
Done

ekohl commented on contribute.md:
````diff
@@ -97,48 +97,52 @@ In general, any Ruby version supported by Rails version required by Foreman will
 ```
-dnf module enable nodejs:14
+# (OPTIONAL) Change Ruby module version before installing
+dnf module enable nodejs:18
````
This isn't optional on EL 9 because by default it installs NodeJS 16.

d1nuc0m replied:
Done

ekohl commented on contribute.md:
```diff
@@ -97,48 +97,52 @@ In general, any Ruby version supported by Rails version required by Foreman will
 #### Supported PostgreSQL versions
 
-In general, the latest version of PostgreSQL will work. Minimum version is currently 10, anything newer than that will do the job.
+Foreman requires PostgreSQL, even for development. SQLite is no longer supported and will not work. In general, the latest version of PostgreSQL will work. Minimum version is currently 13, anything newer than that will do the job.
```
We haven't supported SQLite since Foreman 2.0. Perhaps we can leave that out by now.

d1nuc0m replied:
Ok, 5 years it's enough, removed it :)

ekohl commented on contribute.md:
```diff
@@ -162,19 +166,39 @@ sudo -u postgres createdb foreman
+dnf groupinstall "Development Tools"
+dnf install libvirt-devel postgresql-devel openssl-devel libxml2-devel \
+  sqlite-devel libxslt-devel zlib-devel readline-devel systemd-devel libcurl-devel krb5-devel
+dnf install chromedriver
```
This is in EPEL so I'd at least add a note

d1nuc0m replied:
Done

ekohl commented on contribute.md:
```diff
@@ -210,39 +234,42 @@ Create database tables and seed the initial data. New administrator password, in
 * SEED_ORGANIZATION
 * SEED_LOCATION
 
+Choose an environment from `config/environments` to set RAIL_ENV.
```
```suggestion
Choose an environment from `config/environments` to set RAILS_ENV.
```

d1nuc0m replied:
Done

ekohl commented on contribute.md:
````diff
@@ -210,39 +234,42 @@ Create database tables and seed the initial data. New administrator password, in
 ```
-bundle exec rake db:migrate
-SEED_ADMIN_PASSWORD=changeme bundle exec rake db:seed
+RAILS_ENV=test bundle exec rake db:create
+RAILS_ENV=test bundle exec rake db:migrate
+RAILS_ENV=test bundle exec rake db:seed SEED_ADMIN_PASSWORD=changeme
````
Normally you use the `development` environment (which is default). The `test` environment is really only intended to run tests. Running tests can randomly wipe the DB which is a very frustrating development experience.

d1nuc0m replied:
Done and tried to explain the different environments

By the way, I tried to run some rake tasks without explicitly setting RAILS_ENV but they failed, am I supposed to set RAILS_ENV=development even if it is a default?

d1nuc0m commented:
@ekohl can I also remember of this?

d1nuc0m commented:
@ekohl rebased due to conflicts

maximiliankolb commented on contribute.md:
```diff
@@ -5,39 +5,50 @@ title: Contributing to Foreman
 
-The Foreman is an open-source project that's licensed under the GNU General Public License version 3. Contributions of all types are gladly accepted! Before setting up your development environment, check out our [issue tracker](https://projects.theforeman.org/projects). We have a list of [trivial or easy](https://projects.theforeman.org/projects/foreman/issues?query_id=42) issues in the Foreman c ...
+The Foreman is an open-source project that's licensed under the GNU General Public License version 3.
+Contributions of all types are gladly accepted! Before setting up your development environment, check out our [issue tracker](https://projects.theforeman.org/projects).
```
```suggestion
Contributions of all types are gladly accepted!
Before setting up your development environment, check out our [issue tracker](https://projects.theforeman.org/projects).
```

d1nuc0m replied:
Done

maximiliankolb commented on contribute.md:
```diff
@@ -5,39 +5,50 @@ title: Contributing to Foreman
+  This can be used for evaluation or pilot installation.
+
+You can substitute `centos9-stream` with `centos10-stream` or any tag from files in [forklift/vagrant/boxes.d](https://github.com/theforeman/forklift/tree/master/vagrant/boxes.d).
+Similarly, `nightly` can be replaced by `2.4` or any other supported version. Instead of `foreman`, `katello` can also be used.
```
```suggestion
Similarly, `nightly` can be replaced by `2.4` or any other supported version.
Instead of `foreman`, `katello` can also be used.
```

Another idea: maybe bump the example version to 3.17? I know that it will be outdated some day again, but still.

d1nuc0m replied:
Done

maximiliankolb commented on contribute.md:
```diff
@@ -62,9 +75,12 @@ npm install
+It is also possible to clone and enable plugin repositories, however we do not recommend to manually install the Katello plugin which require number of backend systems to be installed and configured.
 
-This chapter assumes a Linux OS preferably with Fedora or CentOS 8. For other Linux distributions, some instructions might differ. If you are on MacOS, there is an unofficial [guide](https://community.theforeman.org/t/foreman-development-setup-on-mac/22906) available. Windows is not supported as Foreman requires UNIX-compabitle OS.
+This chapter assumes a Linux OS preferably with Fedora or CentOS 8. For other Linux distributions, some instructions might differ.
```
```suggestion
This chapter assumes a Linux OS preferably with Fedora or CentOS 8.
For other Linux distributions, some instructions might differ.
```

d1nuc0m replied:
Done

maximiliankolb commented on contribute.md:
```diff
@@ -374,9 +451,8 @@ Please make sure there is a [Redmine issue](/contribute.html#Bugreporting) open
-*  Make changes and commit. Please reference the Redmine issue this commit
-addresses via "Refs" or "Fixes" in the commit message. See [Coding
-Standards](handbook.html#Codingstandards) guide for more details.
+*  Make changes and commit. Please reference the Redmine issue this commit addresses via "Refs" or "Fixes" in the commit message.
```
```suggestion
* Make changes and commit. Please reference the Redmine issue this commit addresses via "Refs" or "Fixes" in the commit message.
```

d1nuc0m replied:
Done

maximiliankolb reviewed (commented):
Thanks!

I focused my review on the change to use one sentence per line. Looks good overall; I made four minor suggestions. I cannot comment much on some of the technical changes but everything looks reasonable to me.

maximiliankolb reviewed (approved):
LGTM
